src/tiled/scriptmodule.cpp

```diff
@@ -282,6 +306,63 @@ QJSValue ScriptModule::registerTool(const QString &shortName, QJSValue toolObjec
     return toolObject;
 }
 
+ScriptMapFormatWrapper *ScriptModule::mapFormat(const QString &shortName) const
+{
+    const auto formats = PluginManager::objects<MapFormat>();
+    for (auto format : formats) {
+        if (format->shortName() == shortName) {
+            auto wrapper = new ScriptMapFormatWrapper(format);
+            QQmlEngine::setObjectOwnership(wrapper, QQmlEngine::JavaScriptOwnership);
```

Comment:
It occurred to me that in this case, explicitly setting the ownership is not needed, because you're returning a QObject without parent from a function, which by default will have JavaScript ownership.